### PR TITLE
Add HereID and HereCommon AP_Periph modules

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/HereID/hwdef-bl.dat [hunk 1]
```diff
@@ -0,0 +1,9 @@
+include ../HereCommon/hwdef-bl.inc
+
+# board ID for firmware load
+APJ_BOARD_ID 1038
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/HereID/hwdef-bl.dat:4]: fix id

File: libraries/AP_HAL_ChibiOS/hwdef/HereID/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,10 @@
+include ../HereCommon/hwdef.inc
+
+# board ID for firmware load
+APJ_BOARD_ID 1042
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/HereID/hwdef.dat:4]: fix id

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

File: Tools/ardupilotwaf/boards.py [hunk 3]
```diff
@@ -425,6 +425,8 @@ def get_ap_periph_boards():
     for d in dirlist:
         if d in list_ap:
             continue
+        if 'Here' in d:
```
Comment (peterbarker) [inline Tools/ardupilotwaf/boards.py:428]: Oh, I don't like this....  let's not sprinkle product names through the actual code, please keep it in data as much as possible,

**Review by peterbarker — CHANGES_REQUESTED** [review]
Should we pull that change for the CPU architecture out into a separate PR for separate consideration?

File: Tools/ardupilotwaf/boards.py (hunk 3 above)
Comment (tridge) [reply Tools/ardupilotwaf/boards.py:428]: it should find AP_PERIPH in the hwdef.dat, so this should not be needed?

Comment (bugobliterator) [reply Tools/ardupilotwaf/boards.py:428]: AP_Periph define is in the .inc hwdef not the .dat.

Comment (bugobliterator) [reply Tools/ardupilotwaf/boards.py:428]: We plan on having Here in the name of all our AP_Periph products, they will include HereCommon/hwdef.inc . So, most if not all the hwdef based on them will not have AP_Periph in the hwdef, that is why I had to make that change.

**Review by peterbarker — APPROVED** [review]

**Pull request merged**